src/main/java/com/barterbay/app/servcie/GoodService.java

```diff
@@ -0,0 +1,50 @@
+package com.barterbay.app.servcie;
+
+import com.barterbay.app.domain.Good;
+import com.barterbay.app.domain.dto.good.GoodCreatedDTO;
+import com.barterbay.app.mapper.AddressMapper;
+import com.barterbay.app.mapper.GoodMapper;
+import com.barterbay.app.repository.GoodRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+@AllArgsConstructor
+public class GoodService {
+
+  private final GoodRepository goodRepository;
+  private final GoodMapper goodMapper;
+  private final AddressMapper addressMapper;
+
+  @Transactional(readOnly = true)
+  public GoodCreatedDTO findGoodById(long goodId) {
+    return goodMapper.goodEntityToGoodDTO(goodRepository.findById(goodId).orElseThrow());
```

Comment:
`orElseThrow()` - or else throw what? 
In case you can not find goods by id -  what customer will see? 100 line stack trace?
Should be thrown your custom exception, and that exception should be handled